flink-python/pyflink/fn_execution/datastream/window/window_assigner.py

```diff
@@ -0,0 +1,376 @@
+        # No action is taken on the window.
+        return TriggerResult.CONTINUE
+
+    def on_event_time(self,
+                      time: int,
+                      window: CountWindow,
+                      ctx: Trigger.TriggerContext) -> TriggerResult:
+        # No action is taken on the window.
+        return TriggerResult.CONTINUE
+
+    def on_merge(self, window: CountWindow, ctx: Trigger.OnMergeContext) -> None:
+        ctx.merge_partitioned_state(self._count_state_descriptor)
+
+    def clear(self, window: CountWindow, ctx: Trigger.TriggerContext) -> None:
+        state = ctx.get_partitioned_state(self._count_state_descriptor)
+        state.clear()
+
+
+class TumblingWindowAssigner(WindowAssigner[object, TimeWindow]):
+    """
+    A WindowAssigner that windows elements into windows based on the current system time of the machine the operation is running on. Windows cannot overlap.
+    For example, in order to window into windows of 1 minute, every 10 seconds:
+    ::
+            >>> data_stream.key_by(lambda x: x[0], key_type=Types.STRING()) \
+            >>> .window(TumblingWindowAssigner(Time.minutes(1), Time.seconds(10), False))
+
+    A WindowAssigner that windows elements into windows based on the timestamp of the elements. Windows cannot overlap.
+    For example, in order to window into windows of 1 minute:
+     ::
+            >>> data_stream.assign_timestamps_and_watermarks(watermark_strategy) \
+            >>> .key_by(lambda x: x[0], key_type=Types.STRING()) \
+            >>> .window(TumblingWindowAssigner(Time.minutes(1), Time.seconds(0), True))
+    """
+    def __init__(self, size: Time, offset: Time, is_event_time: bool):
+        self._size = size.to_milliseconds()
+        self._offset = offset.to_milliseconds()
+        self._is_event_time = is_event_time
+
+    def assign_windows(self,
+                       element: object,
+                       timestamp: int,
+                       context: WindowAssigner.WindowAssignerContext) -> Collection[TimeWindow]:
+        if self._is_event_time is False:
+            timestamp = context.get_current_processing_time()
+
+        start = TimeWindow.get_window_start_with_offset(timestamp, self._offset, self._size)
+        return [TimeWindow(start, start + self._size)]
+
+    def get_default_trigger(self, env) -> Trigger[T, W]:
+        if self._is_event_time is True:
+            return EventTimeTrigger()
+        else:
+            return ProcessingTimeTrigger()
+
+    def get_window_serializer(self) -> TypeSerializer[W]:
+        return TimeWindowSerializer()
+
+    def is_event_time(self) -> bool:
+        return self._is_event_time
+
+    def __repr__(self):
+        return "TumblingWindowAssigner(%s,%s,%s)" % (self._size, self._offset, self.is_event_time)
+
+'''
+A WindowAssigner that windows elements into windows based on the number of the elements. Windows cannot overlap.
+'''
+class CountTumblingWindowAssigner(WindowAssigner[object, CountWindow]):
```

Comment:
It uses GlobalWindows for count window in the Java API. Is it possible to do so in Python?